src/client/serialize/daos_serialize.c

```diff
@@ -8,13 +8,88 @@
 
 /* for user attr dataset */
 struct usr_attr {
-	char	*attr_name;
-	hvl_t	attr_val;
+    char    *attr_name;
+    hvl_t   attr_val;
 };
 
+/* struct obj_id type defined for hdf5 attribute. The DAOS_PROP_CO_ROOTS
+ * points to an array of daos_obj_id_t, which is a struct that contains
+ * hi, lo uint64_t's. In order to write these to an hdf5 file a similar
+ * type has to be defined to describe/store it to hdf5
+ */
+struct obj_id {
+    uint64_t hi;
+    uint64_t lo;
+};
```

Comment:
@daltonbohning I tried and it didn't work when I did try to use it directly. This is a definition for inside of the hdf5 file, which doesn't recognize the daos_obj_id_t type in the file. This is why the acl is actually written as a list of strings because you cannot write to the file as a daos_acl type, hdf5 does not know how that is represented in DAOS.

Reply:
@dsikich `daos_prop_t` makes sense, because `daos_prop_t.dpp_entries` is an array. But here, `struct obj_id` is almost the same as `daos_obj_id_t`, and `daos_obj_id_t` is public, so I would think that should work. But I'm not familiar enough with HDF5

Reply:
@daltonbohning oh oops I see what you mean use the type that was defined to write the OIDS, yeah I can probably do that. I misunderstood and that you meant the native DAOS type.